internal/daemon/controller/handlers/list_token.go

```diff
@@ -8,42 +8,80 @@ import (
 
 	"github.com/hashicorp/boundary/internal/errors"
 	pbs "github.com/hashicorp/boundary/internal/gen/controller/api/services"
+	"github.com/hashicorp/boundary/internal/listtoken"
 	"github.com/hashicorp/boundary/internal/types/resource"
 	"github.com/mr-tron/base58"
 	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-// ParseRefreshToken parses a refresh token from the input, returning
+// ParseListToken parses a list token from the input, returning
 // an error if the parsing fails.
-func ParseRefreshToken(ctx context.Context, token string) (*pbs.ListRefreshToken, error) {
-	const op = "handlers.ParseRefreshToken"
+func ParseListToken(ctx context.Context, token string) (*pbs.ListToken, error) {
+	const op = "handlers.ParseListToken"
 	marshaled, err := base58.Decode(token)
 	if err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
-	var tok pbs.ListRefreshToken
+	var tok pbs.ListToken
 	if err := proto.Unmarshal(marshaled, &tok); err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
 	return &tok, nil
 }
 
-// MarshalRefreshToken marshals a refresh token to its string representation.
-func MarshalRefreshToken(ctx context.Context, token *pbs.ListRefreshToken) (string, error) {
-	const op = "handlers.MarshalRefreshToken"
+// MarshalListToken assembles and marshals a list token to its string representation.
+func MarshalListToken(ctx context.Context, token *listtoken.Token, resourceType pbs.ResourceType) (string, error) {
+	const op = "handlers.MarshalListToken"
```

Comment:
why not check for an unspecified resourceType?  using a switch?

Reply:
Good idea